etl/steps/data/meadow/imf/2023-11-02/world_economic_outlook.py

```diff
@@ -0,0 +1,96 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import numpy as np
+from owid.catalog import Table
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+# This is a preliminary bulk importer for the IMF's WEO dataset.
+# As of October 2021, it only generates a grapher-compatible dataset with 1 variable (GDP growth).
+# But this first version could be extended to a traditional bulk import of the entire dataset later.
+VARIABLE_LIST = [
+    "NGDP_RPCH",  # Gross domestic product, constant prices / Percent change
+    "LUR",  # Unemployment rate / Percent of total labor force
+]
+
+
+def select_data(tb: Table) -> Table:
+    tb = tb.drop(
+        columns=[
+            "WEO Country Code",
+            "ISO",
+            "Country/Series-specific Notes",
+            "Subject Notes",
+            "Scale",
+        ]
+    ).dropna(subset=["Country"])
+    return tb
+
+
+def make_variable_names(tb: Table) -> Table:
+    tb["variable"] = tb["Subject Descriptor"] + " - " + tb["Units"]
+    tb = tb.drop(columns=["Subject Descriptor", "Units"])
+    return tb
+
+
+def pick_variables(tb: Table) -> Table:
+    return tb[tb["WEO Subject Code"].isin(VARIABLE_LIST)].drop(columns="WEO Subject Code")
+
+
+def reshape_and_clean(tb: Table) -> Table:
+    tb = tb.melt(id_vars=["Country", "variable", "Estimates Start After"], var_name="year")
+
+    # Coerce values to numeric.
+    tb["value"] = tb["value"].replace("--", np.nan).astype(float)
+    tb["year"] = tb["year"].astype(int)
+
+    # Split between observations and forecasts
+    tb.loc[tb.year > tb["Estimates Start After"], "variable"] += "_forecast"
+    tb.loc[tb.year <= tb["Estimates Start After"], "variable"] += "_observation"
+
+    # Drop rows with missing values.
+    tb = tb.dropna(subset=["value"])
+
+    # Drop Estimates Start After
+    tb = tb.drop(columns="Estimates Start After")
+
+    tb = tb.pivot(index=["Country", "year"], columns="variable", values="value", join_column_levels_with="_")
+    return tb
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("world_economic_outlook.xls")
+
+    # Load data and metadata from snapshot.
+    tb = snap.read_csv(delimiter="\t", encoding="ISO-8859-1")
+
+    #
+    # Process data.
+    #
+    # Prepare raw data.
+    tb = select_data(tb).pipe(make_variable_names).pipe(pick_variables).pipe(reshape_and_clean)
```

Comment:
My personal opinion (no need to change anything): I think `pipe` should be avoided. It reduces the amount of code inside the `run` function and moves all useful comments to other places in the script (usually comments inside functions or the docstrings of the functions). With this, we lose readability.
The pipeline is much easier to follow if important things happen line after line with a short comment, rather than everything in one line without any comments.

Reply:
There are too extremes in terms of how explicit the code should be:
1. In one extreme, you only have a `run` function, and it's a really long piece of spaghetti code, where it's really hard to easily understand the main things that happen to the data.
2. In the other extreme, the `run` function contains only one line, which calls another function called `process_data()`, and if anyone wants to understand what happens there, they need to dig into that function.

There's a trade-off, but the metric to optimize is readability. For small datasets like this, we definitely don't care about performance. For things like FAOSTAT we may sometimes need to optimize for performance, at the cost 
 of readability (to avoid memory exploding, or extremely long execution times).